cmake/scripts/common/ModuleHelpers.cmake

```
@@ -0,0 +1,34 @@
+# This script provides helper functions for FindModules
+
+# Get archive name from VERSION dependency file
+# Arguments:
+#   name name of the library
+# On return:
+#   ARCHIVE will be set to parent scope
+#   MODULENAME_VER will be set to parent scope (eg FFMPEG_VER, DAV1D_VER)
+#   BASE_URL will be set to parent scope if in VERSION file
+function(get_archive_name module_name)
+  string(TOUPPER ${module_name} UPPER_MODULE_NAME)
+
+  # Dependency path
+  set(${UPPER_MODULE_NAME}_FILE "${CMAKE_SOURCE_DIR}/tools/depends/target/${module_name}/${UPPER_MODULE_NAME}-VERSION")
+
+  file(STRINGS ${${UPPER_MODULE_NAME}_FILE} ${UPPER_MODULE_NAME}_LNAME REGEX "^[ \t]*LIBNAME=")
+  file(STRINGS ${${UPPER_MODULE_NAME}_FILE} ${UPPER_MODULE_NAME}_VER REGEX "^[ \t]*VERSION=")
+  file(STRINGS ${${UPPER_MODULE_NAME}_FILE} ${UPPER_MODULE_NAME}_ARCHIVE REGEX "^[ \t]*ARCHIVE=")
+  file(STRINGS ${${UPPER_MODULE_NAME}_FILE} ${UPPER_MODULE_NAME}_BASE_URL REGEX "^[ \t]*BASE_URL=")
+
+  string(REGEX REPLACE ".*LIBNAME=([^ \t]*).*" "\\1" ${UPPER_MODULE_NAME}_LNAME "${${UPPER_MODULE_NAME}_LNAME}")
+  string(REGEX REPLACE ".*VERSION=([^ \t]*).*" "\\1" ${UPPER_MODULE_NAME}_VER "${${UPPER_MODULE_NAME}_VER}")
+  string(REGEX REPLACE ".*ARCHIVE=([^ \t]*).*" "\\1" ${UPPER_MODULE_NAME}_ARCHIVE "${${UPPER_MODULE_NAME}_ARCHIVE}")
+  string(REGEX REPLACE ".*BASE_URL=([^ \t]*).*" "\\1" ${UPPER_MODULE_NAME}_BASE_URL "${${UPPER_MODULE_NAME}_BASE_URL}")
+
+  string(REGEX REPLACE "\\$\\(LIBNAME\\)" "${${UPPER_MODULE_NAME}_LNAME}" ${UPPER_MODULE_NAME}_ARCHIVE "${${UPPER_MODULE_NAME}_ARCHIVE}")
+  string(REGEX REPLACE "\\$\\(VERSION\\)" "${${UPPER_MODULE_NAME}_VER}" ${UPPER_MODULE_NAME}_ARCHIVE "${${UPPER_MODULE_NAME}_ARCHIVE}")
+
+  set(ARCHIVE ${${UPPER_MODULE_NAME}_ARCHIVE} PARENT_SCOPE)
+  set(${UPPER_MODULE_NAME}_VER ${${UPPER_MODULE_NAME}_VER} PARENT_SCOPE)
+  if (NOT ${UPPER_MODULE_NAME}_BASE_URL STREQUAL "")
```

I think you can just do (not 100%)
```suggestion
  if (NOT ${UPPER_MODULE_NAME}_BASE_URL)
```